From e5707c4f7e88d483426ab59dc0da314711203a60 Mon Sep 17 00:00:00 2001
From: Jai Luthra <j-luthra@ti.com>
Date: Mon, 28 Aug 2023 18:18:40 +0530
Subject: [PATCH 003/322] media: cadence: csi2rx: Fix check for running bit

The running bit (31) of STREAM_STATUS register should go to idle (0)
when stream stop operation is finished. That was the intention in the
original patch but we were checking for it to be running (1) due to a
typo.

Fixes: c4cae42493cb ("media: cadence: csi2rx: Set the STOP bit when stopping a stream")
Signed-off-by: Jai Luthra <j-luthra@ti.com>
Signed-off-by: Vignesh Raghavendra <vigneshr@ti.com>
---
 drivers/media/platform/cadence/cdns-csi2rx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/cadence/cdns-csi2rx.c b/drivers/media/platform/cadence/cdns-csi2rx.c
index 4afb2ce351a0..c6d9aaf38184 100644
--- a/drivers/media/platform/cadence/cdns-csi2rx.c
+++ b/drivers/media/platform/cadence/cdns-csi2rx.c
@@ -410,7 +410,7 @@ static void csi2rx_stop(struct csi2rx_priv *csi2rx)
 		ret = readl_relaxed_poll_timeout(csi2rx->base +
 						 CSI2RX_STREAM_STATUS_REG(i),
 						 val,
-						 (val & CSI2RX_STREAM_STATUS_RDY),
+						 !(val & CSI2RX_STREAM_STATUS_RDY),
 						 10, 10000);
 		if (ret)
 			dev_warn(csi2rx->dev, "Failed to stop stream%u\n", i);
-- 
2.43.0

